### DISCUSSION TITLE
Format of the `serverProperties` table

### DISCUSSION THREAD

**Discussion by trmckay** [scalameta/nvim-metals#512] [Q&A] [answered]
Hi, I am trying to set a server property equivalent to `-Dmetals.bloop-port=<N>` from my Lua nvim-metals config. I have two questions about this.

1. Do I need to pass this to `serverProperties` as a nested table, e.g.

```lua
metals_config.init_options = {
    serverProperties = {
        ["metals"] = {
            ["bloop-port"] = 8000,
        }
    },
},
```

or just as one key in the `serverProperties` table, e.g.

```lua
metals_config.init_options = {
    serverProperties = {
        ["metals.bloop-port"]  = 8000,
    },
},
```

2. Does the type of the value matter? I am using a `number` here, but I'm not sure if I should us a `string` instead.

**Comment by ckipp01** [comment, marked as answer]
Hey @trmckay `serverProperties` isn't nested inside of `init_options`, it's just inside of the config table. So it'd be like:

```lua
metals_config = {
    serverProperties = {
        "-Dmetals.bloop-port=8000",
    },
},
```
Also, this setting is sort of hidden primarily because it's rarely ever used. I've actually never seen someone ask about it. What's your use case here that you want to manually set this?

**Reply by trmckay** [reply]
Thanks! My use case is admittedly niche. I share a machine with other Metals users, so I need to make sure to pick a port that is not in use.
